### ISSUE TITLE
A list of all the properties supported in Spring Boot should be available

### ISSUE THREAD

**Issue by mrioan** [stormpath/stormpath-sdk-java#897]
We could somehow create it out of the `metadata.json` file. We need to be sure that this son file is always up to date. Additionally, a test should validate this during every build.

**Comment by mraible** [comment]
@lhazlewood What's the end goal of this effort? 

I believe we can ensure our `metadata.json` properties match what `@Value`s can be configured and vice-versa. Then we could write something that converts the `metadata.json` into an HTML file that can be included in the Java SDK docs. Does that sound reasonable?

**Comment by lhazlewood** [comment]
@mraible that sounds awesome.  Whatever works and is easiest - for example, maybe the metatadata.json is used to generate a .rst file if that's simpler.

The most important part for me personally is that the build fails if a property is added to either stormpath.properties or stormpath.web.properties if it is also not also represented in the metadata.json files, unless the property exists in a whitelist.  That whitelist would be populated with only the properties that are specific to the servlet integration that are not present in the Spring/Boot/Security integrations.

In other words, all properties in stormpath.properties and stormpath.web.properties should be verified to be in metadata.json files as well, _except_ for the ones that we indicate are specific only to the servlet integration.

Does this make sense?
